### PR TITLE
Export tests with annotations

### PR DESCRIPTION
**Pull request by rebeccacremona**
This is a first pass at testing `.docx` export with annotations. Closes https://github.com/harvard-lil/h2o/issues/755

I left a fair bit of work undone:
- I created my own very basic annotation fixtures for the purpose. While developing, I commented out the already-present annotation fixtures and ran the full test suite: all tests passed anyway, with them absent. I did not investigate. But, this suggests that there's some tidying to be done
- I did not attempt to add any interesting annotations: the existing case text contained a single paragraph with text reading "content to elide" and similar, so I just followed instructions. Soon-ish, we should add better annotation fixtures, covering some of our more complex cases (multi-paragraph elisions, annotations that do fun things with whitespace ala https://github.com/harvard-lil/h2o/issues/680, etc.)
- ~We should probably tweak the test design so that the same casebook/section/resource can be exported with and without annotations.~ just did this.... because I introduced a bug, and this would have caught it......

### CODE REVIEW

File: test/system/export_test.rb [hunk 1]
```diff
@@ -25,27 +25,41 @@ def teardown
...
 
-  def export_test path, format, file
+  def export_test path, format, file, include_annotations
     sign_in user = users(:verified_student)
     visit path
-    click_link 'Export'
 
+    if include_annotations
+      click_link 'Export'
+      click_button 'Without annotations'
```
Comment (cgruppioni) [inline test/system/export_test.rb]: I'm shocked that you got clicking in a modal to work so easily. I feel like I've tried this in the past and had to try a bunch of things and was unsuccessful 🤷‍♀️

File: app/assets/javascripts/lib/ui/content/export_modal.js.erb [hunk 2]
```diff
@@ -26,35 +27,25 @@ function export_casebook_path(casebookId, format = 'docx') {
...
-  let includeAnnotations = <%= Rails.application.config.export_annotations_by_default %>;
   if (pageInfo.resourceId)  {
-    window.location.assign(resource_export_path(pageInfo.resourceId) + (includeAnnotations ? '': '?annotations=false'));
+    window.location.assign(resource_export_path(pageInfo.resourceId) + (includeAnnotations ? '?annotations=true' : '?annotations=false'));
   } else if (pageInfo.sectionId)  {
-    window.location.assign(section_export_path(pageInfo.sectionId) + (includeAnnotations ? '' : '?annotations=false'));
+    window.location.assign(section_export_path(pageInfo.sectionId) + (includeAnnotations ? '?annotations=true' : '?annotations=false'));
   } else {
-    window.location.assign(export_casebook_path(pageInfo.casebookId) + (includeAnnotations ? '' : '?annotations=false'));
+    window.location.assign(export_casebook_path(pageInfo.casebookId) + (includeAnnotations ? '?annotations=true' : '?annotations=false'));
```
Comment (cgruppioni) [inline app/assets/javascripts/lib/ui/content/export_modal.js.erb]: Great point with the window.location.assign

File: test/system/export_test.rb (hunk 1 above)
Comment (rebeccacremona) [reply test/system/export_test.rb]: Wait a minute... if `include annotations` click `without annotations`????? Something is wrong here.

Comment (cgruppioni) [reply test/system/export_test.rb]: Ahhhh yes

**Comment by rebeccacremona** [comment]
Okay: reopening.

Now this tests all three possibilities: export of things without any annotations, export of annotated things using the "with" option and using the "without" option.

If I had done that the first time around, I would have noticed that I introduced a bug: `includeAnnotations` is a string, not a `Boolean` and would have noticed that same bug in the existing ruby-interpolated default value.

This should be all set now. 

If you are comfortable with this version, @casey, please merge at your convenience.

**Comment by rebeccacremona** [comment]
(public shame on me for not writing all 6 tests the first time)

File: app/controllers/content/sections_controller.rb [hunk 3]
```diff
@@ -109,8 +109,9 @@ def reorder
 
   private
 
-  def export_filename format
-    helpers.truncate(@section.title, length: 45, omission: '-', separator: ' ') + '.' + format
+  def export_filename format, annotations=false
```
Comment (cgruppioni) [inline app/controllers/content/sections_controller.rb]: You can move this method and the section & casebook one into 1 method in the node model. 

Ex: 
`@section.export_filename('docx', @include_annotations)`

**Comment by cgruppioni** [comment]
@rebeccacremona Just one small thing I commented on

Also, I usually like to make a test plan for whoever does it on staging. Just something simple like -- 

Testing Plan: 
- [ ] Take a resource, section and casebook that has annotations and export them with `with annotations` selected
- [ ] Take a resource, section and casebook that has annotations and export them with `no annotations` selected
- [ ] Take a resource, section and casebook that doesn't have annotations and export them
Make sure exports export successfully, inside content should not be changed by this ticket 

Do you think this covers everything that could be effects?

File: app/controllers/content/sections_controller.rb (hunk 3 above)
Comment (rebeccacremona) [reply app/controllers/content/sections_controller.rb]: Yeah, it seems goofy to have duplicate export logic in each of the three classes. Is it okay with you if I do not refactor that existing export logic as part of this PR? I just added a parameter to already-existing methods in the three classes. If you WOULD like that refactoring to happen at the same time, no problem, I'll close this and can add that later this week or early next week.

**Comment by rebeccacremona** [comment]
@cgruppioni Yep, that sounds right, thank you for writing it out! I can definitely start including steps for manual testing in future PRs. (FWIW, I usually pop over to stage as soon as something gets merged to have a check personally.)

> Do you think this covers everything that could be effects?

File: app/controllers/content/sections_controller.rb (hunk 3 above)
Comment (cgruppioni) [reply app/controllers/content/sections_controller.rb]: That's fine!

**Comment by cgruppioni** [comment]
Just tested, good to be merged in tomorrow

**Pull request merged**